tapgarden/planter.go

```diff
@@ -1273,47 +1284,116 @@ func (c *ChainPlanter) sealBatch(ctx context.Context, _ SealParams) error {
 		c.pendingBatch.GenesisPacket.Pkt.UnsignedTx,
 	)
 
+	// Check if the batch is already sealed by picking a random grouped
+	// seedling and trying to fetch the full asset group.
+	var singleSeedling []*Seedling
+	for _, seedling := range groupSeedlings {
+		singleSeedling = append(singleSeedling, seedling)
+		break
+	}
+
+	// If the batch was previously sealed, each grouped seedling will have
+	// its asset genesis already stored on disk.
+	existingGroups, err := c.cfg.Log.FetchSeedlingGroups(
+		ctx, genesisPoint, anchorOutputIndex, singleSeedling,
+	)
+
+	switch {
+	case len(existingGroups) != 0:
+		return nil, fmt.Errorf("batch is already sealed")
+	case err != nil:
+		// The only expected error is for a missing asset genesis.
+		errMsg := err.Error()
+		if !strings.Contains(errMsg, "unable to fetch genesis asset") {
```

Comment:
Feels like we should return a typed error here and use `errors.Is()` instead of string matching...

Reply:
I think I tried that before, but pulling in the concrete error from `tapdb` causes an import cycle.